lib/reboot_cmds.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+"""
+Some uaclient operations cannot be fully completed by running a single
+command. For example, when upgrading uaclient from trusty to xenial,
+we may have a livepatch change in the contract, allowing livepatch to be
+enabled on xenial. However, during the upgrade we cannot install livepatch on
+the system, only after a reboot.
+
+To allow uaclient to postpone commands that need to be executed in a system boot,
+we are using this script, which will basically try to reprocess contract deltas
+on boot time.
+"""
+import logging
+import os
+import sys
+
+from uaclient import config
+
+from uaclient.util import subp, ProcessExecutionError
+from uaclient.cli import setup_logging, assert_lock_file
+
+
+def run_command(cmd, cfg):
+    try:
+        out, _ = subp(cmd.split(), capture=True)
+        logging.debug("Successfully executed cmd: {}".format(cmd))
+    except ProcessExecutionError as exec_error:
+        msg = (
+            "Failed running cmd: {}\n"
+            "Return code: {}\n"
+            "Stderr: {}\n"
+            "Stdout: {}".format(
+                cmd, exec_error.exit_code, exec_error.stderr, exec_error.stdout
+            )
+        )
+
+        reboot_cmd_marker_file = cfg.data_path("marker-reboot-cmds")
+        cfg.delete_cache_key(reboot_cmd_marker_file)
```

Comment:
we actually need to use the UAConfig data_path key here, not a file path
```suggestion
        cfg.delete_cache_key("marker-reboot-cmds")
```